numpy/lib/user_array.py

```diff
@@ -1,286 +1 @@
-            return func(self[0])
-        else:
-            raise TypeError(
-                "only rank-0 arrays can be converted to Python scalars.")
-
-    def __complex__(self):
-        return self._scalarfunc(complex)
-
-    def __float__(self):
-        return self._scalarfunc(float)
-
-    def __int__(self):
-        return self._scalarfunc(int)
-
-    def __hex__(self):
-        return self._scalarfunc(hex)
-
-    def __oct__(self):
-        return self._scalarfunc(oct)
-
-    def __lt__(self, other):
-        return self._rc(less(self.array, other))
-
-    def __le__(self, other):
-        return self._rc(less_equal(self.array, other))
-
-    def __eq__(self, other):
-        return self._rc(equal(self.array, other))
-
-    def __ne__(self, other):
-        return self._rc(not_equal(self.array, other))
-
-    def __gt__(self, other):
-        return self._rc(greater(self.array, other))
-
-    def __ge__(self, other):
-        return self._rc(greater_equal(self.array, other))
-
-    def copy(self):
-        ""
-        return self._rc(self.array.copy())
-
-    def tostring(self):
-        ""
-        return self.array.tostring()
-
-    def tobytes(self):
-        ""
-        return self.array.tobytes()
-
-    def byteswap(self):
-        ""
-        return self._rc(self.array.byteswap())
-
-    def astype(self, typecode):
-        ""
-        return self._rc(self.array.astype(typecode))
-
-    def _rc(self, a):
-        if len(shape(a)) == 0:
-            return a
-        else:
-            return self.__class__(a)
-
-    def __array_wrap__(self, *args):
-        return self.__class__(args[0])
-
-    def __setattr__(self, attr, value):
-        if attr == 'array':
-            object.__setattr__(self, attr, value)
-            return
-        try:
-            self.array.__setattr__(attr, value)
-        except AttributeError:
-            object.__setattr__(self, attr, value)
-
-    # Only called after other approaches fail.
-    def __getattr__(self, attr):
-        if (attr == 'array'):
-            return object.__getattribute__(self, attr)
-        return self.array.__getattribute__(attr)
-
-#############################################################
-# Test of class container
-#############################################################
-if __name__ == '__main__':
-    temp = reshape(arange(10000), (100, 100))
-
-    ua = container(temp)
-    # new object created begin test
-    print(dir(ua))
-    print(shape(ua), ua.shape)  # I have changed Numeric.py
-
-    ua_small = ua[:3, :5]
-    print(ua_small)
-    # this did not change ua[0,0], which is not normal behavior
-    ua_small[0, 0] = 10
-    print(ua_small[0, 0], ua[0, 0])
-    print(sin(ua_small) / 3. * 6. + sqrt(ua_small ** 2))
-    print(less(ua_small, 103), type(less(ua_small, 103)))
-    print(type(ua_small * reshape(arange(15), shape(ua_small))))
-    print(reshape(ua_small, (5, 3)))
-    print(transpose(ua_small))
+from ._user_array_impl import container
```

Comment:
It just occured to me that we probably want to explicitly import `__doc__` from the `_impl.py` files you've created in the NEP 52 work. That way if someone does e.g. `help(np.lib.user_array)` they'll get the docstring printed out as they would have before the refactor.

Feel free to do that in all the `_impl.py` files you've added or do it in a followup PR.

Reply:
That is true. I added:
```python
from ._<...>_impl import __doc__, ...
```
for these submodules that don't define custom `__getattr__`.